tests/cephadm/test_legacy_services.py

```diff
@@ -0,0 +1,55 @@
+from ceph.ceph_admin import CephAdmin
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def run(ceph_cluster, **kw):
+    """
+    Verify if the cephadm shell tries to infer config from
+    a dummy mon created.
+    Verifies bz:2080242
+
+    Args:
+        ceph_cluster: Ceph cluster object
+        **kw :     Key/value pairs of configuration information to be used in the test.
+
+    Returns:
+        int - 0 when the execution is successful else 1 (for failure).
+    """
+
+    config = kw.get("config")
+
+    cephadm = CephAdmin(cluster=ceph_cluster, **config)
+
+    all_nodes = cephadm.cluster.get_nodes()
+    mon_nodes = cephadm.cluster.get_nodes(role="mon")
+    client_node = cephadm.cluster.get_nodes(role="client")
+    # Exclude the client node
+    mon_nodes.extend(client_node)
+
+    # Create dummy mon on all non-mon nodes
+    for node in set(all_nodes) - set(mon_nodes):
```

Comment:
Instead you can use below code -
```
    for node in cephadm.cluster.get_nodes():
        if node.role == "mon":
            continue
        
        <rest of logic>
        .
        .
```

Reply:
In this case, lets say there are around 10 nodes, the loop iterates for 10 times, whereas the current logic will iterate only for total_nodes -(no. of mon_nodes + client nodes) times, which I feel is the efficient way. Please let me know your thoughts on this.

Reply:
To get nodes specific to role `get_nodes` [1] does same think. In this case, for 10 nodes, `get_nodes` will iterate 10 times for role `mon` and 10 times for `client` which will sum to 20.

[1] https://github.com/red-hat-storage/cephci/blob/master/ceph/ceph.py#L101

Reply:
also as per the dummy mon node validation test it is against **only one non-mon node** as of now it's testing on all non-mon nodes which I feel is redundant.
you can add break

Reply:
+1

Reply:
> also as per the dummy mon node validation, test it against **only one non-mon node** as of now it's testing on all non-mon nodes which I feel is redundant. you can add break

+1

Reply:
Ack. Noted. Thanks for the pointer @vamahaja @sunilangadi2 Will make the chnages